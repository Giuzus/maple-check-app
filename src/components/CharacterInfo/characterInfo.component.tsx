import { Character } from "../../store/character.store";

type CharacterInfoProps = {
  character: Character;
};

function CharacterInfo({ character }: CharacterInfoProps) {
  return (
    <div className="flex flex-col">
      <span>{character.name}</span>

      {!character.error ? (
        character.data ? (
          <>
            <span>{character.data?.Class}</span>
            <span>{character.data?.Level}</span>
            <span>{character.data?.Server}</span>
          </>
        ) : (
          <span>Loading...</span>
        )
      ) : (
        <span className="text-red-500">Error loading character data</span>
      )}
    </div>
  );
}

export default CharacterInfo;
