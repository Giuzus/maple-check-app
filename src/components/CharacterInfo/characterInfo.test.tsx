import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import CharacterInfo from "./characterInfo.component";

describe("Character info", () => {
  test("should match snapshot", () => {
    //Arrange
    const page = render(<CharacterInfo character={{ name: "test" }} />);

    //Assert
    expect(page.baseElement).toMatchSnapshot();
  });
});
