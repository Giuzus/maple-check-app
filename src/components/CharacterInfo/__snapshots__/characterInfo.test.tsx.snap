// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Character info > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <span>
        test
      </span>
      <span>
        Loading...
      </span>
    </div>
  </div>
</body>
`;
